lib/rex/job.rb

```diff
@@ -20,6 +20,18 @@ def initialize(container, jid, name, ctx, run_proc, clean_proc)
     self.clean_proc = clean_proc
     self.ctx        = ctx
     self.start_time = nil
+    self.cleanup_mutex = Mutex.new
+    self.cleaned_up = false
+  end
+
+  def synchronized_cleanup
+    # Avoid start and stop both calling cleanup
+    self.cleanup_mutex.synchronize do
+      unless cleaned_up
+        self.clean_proc.call(ctx) if self.clean_proc
+        self.cleaned_up = true
```

Comment:
I think we should add an `ensure` here, so we don't attempt to cleanup multiple times? 👀